### PR TITLE
Add option to add to top rather then bottom

### PR DESCRIPTION
**Pull request by baelish**

### CODE REVIEW

**Comment by baelish** [comment]
I don't think this will work as is, --top could be $1 in which case it all falls down. I'll test it, if it doesn't work I will close the pull request.

File: formulas/wick-base/functions/wick-add-config-section [hunk 1]
```diff
@@ -27,9 +30,18 @@ wickAddConfigSection() {
...
-        echo "${3:-#} BEGIN wickAddConfigSection: $2"
-        cat
-        echo "${3:-#} END wickAddConfigSection: $2"
-    } >> "$1"
+    if [[ -n "$top" ]]; then
+        oldContent=$(cat "$1")
+        {
+            echo "${3:-#} BEGIN wickAddConfigSection: $2"
+            cat
+            echo "${3:-#} END wickAddConfigSection: $2"
```
Comment (kyle-long) [inline formulas/wick-base/functions/wick-add-config-section]: Don't you want to add the `oldContent` here as well?

File: formulas/wick-base/functions/wick-add-config-section [hunk 2]
```diff
@@ -27,9 +30,18 @@ wickAddConfigSection() {
         wickRemoveConfigSection "$1" "$2" "${3-}"
     fi
 
-    {
-        echo "${3:-#} BEGIN wickAddConfigSection: $2"
-        cat
-        echo "${3:-#} END wickAddConfigSection: $2"
-    } >> "$1"
+    if [[ -n "$top" ]]; then
+        oldContent=$(cat "$1")
```
Comment (kyle-long) [inline formulas/wick-base/functions/wick-add-config-section]: `oldContent` is not declared

File: formulas/wick-base/functions/wick-add-config-section [hunk 3]
```diff
@@ -27,9 +30,18 @@ wickAddConfigSection() {
         wickRemoveConfigSection "$1" "$2" "${3-}"
     fi
 
-    {
-        echo "${3:-#} BEGIN wickAddConfigSection: $2"
-        cat
-        echo "${3:-#} END wickAddConfigSection: $2"
-    } >> "$1"
+    if [[ -n "$top" ]]; then
```
Comment (kyle-long) [inline formulas/wick-base/functions/wick-add-config-section]: Possible alternate implementation

```
oldContent=$(cat "$1")
section=$(
    echo "${3:-#} BEGIN wickAddConfigSection: $2"
    cat
    echo "${3:-#} END wickAddConfigSection: $2"
)

{
    if [[ -z "$top" ]]; then
        echo "$oldContent"
        echo "$section"
    else
        echo "$section"
        echo "$oldContent"
    fi
} > "$1"
```

Does this make it any easier to see whats going on?

File: formulas/wick-base/functions/wick-add-config-section [hunk 4]
```diff
@@ -6,13 +6,16 @@
...
 # $3 - Line comment delimeter, defaults to "#"
+# --top - Add section to the top (default adds to the bottom).
 # stdin - Lines of configuration
 #
 # This searches the file for a section with the same name and removes it.
 # Next it creates a new section with the given name at the end of the file.
 #
 # Returns 0 on success, 1 on argument validation errors.
 wickAddConfigSection() {
+    wickGetOption top --top "$@"
```
Comment (quantumew) [inline formulas/wick-base/functions/wick-add-config-section]: Can you declare top so shellcheck doesn't complain? `local top`

File: formulas/wick-base/functions/wick-add-config-section (hunk 1 above)
Comment (baelish) [inline formulas/wick-base/functions/wick-add-config-section]: didn't need to as it uses >> but using your more readable version now anyhow.

File: formulas/wick-base/functions/wick-add-config-section (hunk 4 above)
Comment (baelish) [inline formulas/wick-base/functions/wick-add-config-section]: done and more

File: formulas/wick-base/functions/wick-add-config-section (hunk 2 above)
Comment (baelish) [inline formulas/wick-base/functions/wick-add-config-section]: done

File: formulas/wick-base/functions/wick-add-config-section (hunk 3 above)
Comment (baelish) [inline formulas/wick-base/functions/wick-add-config-section]: I like it done.

**Comment by baelish** [comment]
Also fixed my comment of concern that it wouldn't work, all addressed in commit 4816521

**Comment by fidian** [comment]
Looks good to me.

File: formulas/wick-base/functions/wick-add-config-section (hunk 1 above)
Comment (kyle-long) [inline formulas/wick-base/functions/wick-add-config-section]: But this one doesn't use `>>`. In any case, it all looks good.

File: formulas/wick-base/functions/wick-add-config-section [hunk 5]
```diff
@@ -6,30 +6,47 @@
...
-        echo "${3:-#} BEGIN wickAddConfigSection: $2"
+    oldContent=$(cat "${args[0]}")
+    section=$(
+        echo "${args[2]:-#} BEGIN wickAddConfigSection: ${args[1]}"
         cat
-        echo "${3:-#} END wickAddConfigSection: $2"
-    } >> "$1"
+        echo "${args[2]:-#} END wickAddConfigSection: ${args[1]}"
+    )
+    if [[ -z "$top" ]]; then
```
Comment (kyle-long) [inline formulas/wick-base/functions/wick-add-config-section]: The curly brackets can go around the whole if/else block. Everything is going to the same spot.

Comment (baelish) [inline formulas/wick-base/functions/wick-add-config-section]: Agreed e0baeb7

**Comment by baelish** [comment]
Thanks for the reviews all!

**Pull request merged**